packages/tests/e2e/auth.spec.ts

```diff
@@ -60,4 +61,92 @@ test.describe("Authentication", () => {
 
     await expect(formPage.successState).toBeVisible();
   });
+
+  test("Reset password flow", async ({ page, context }) => {
+    const testInbox = createTestInbox();
+    const formPage = new FormPage(page);
+
+    await createUserWithPassword(testInbox.email, "T3stPWD!1");
+    await formPage.load();
+
+    // try login with password
+    await formPage.selectAuthMethod("password");
+    await formPage.enterEmail(testInbox.email);
+    await formPage.submitInitialForm();
+
+    await formPage.resetPassword();
+    // check email
+
+    const resetEmail = await testInbox.getEmailBySubject("Reset your password");
+    expect(resetEmail).toBeDefined();
+    if (!resetEmail) return;
+    const jumpPageResetURL = await testInbox.getJumpPageURL(resetEmail);
+    expect(jumpPageResetURL).toBeDefined();
+    if (!jumpPageResetURL) return;
+
+    // change password on jump page
+    const jumpPageReset = new JumpPage(
+      await context.newPage(),
+      jumpPageResetURL
+    );
+    await jumpPageReset.load();
+    await jumpPageReset.page.bringToFront();
+    await jumpPageReset.enterPassword("T3stPWD!2");
+    await jumpPageReset.enterPasswordConfirm("T3stPWD!2");
+
+    await page.waitForTimeout(1000);
+    await jumpPageReset.submitInitialForm();
+
+    await jumpPageReset.successState.waitFor({
+      state: "visible",
+      timeout: 8000,
+    });
+
+    // log in
+    await formPage.page.bringToFront();
+    await formPage.enterPassword("T3stPWD!2");
+    await formPage.submitAuthenticatingForm();
+
+    // success
+    await expect(formPage.successState).toBeVisible();
+  });
+
+  test("Submit incorrect OTP", async ({ page }) => {
+    const testInbox = createTestInbox();
+    const formPage = new FormPage(page);
+
+    await formPage.load();
+
+    await formPage.selectAuthMethod("otp_via_email");
+    await formPage.enterEmail(testInbox.email);
+    await formPage.submitInitialForm();
+
+    const email = await testInbox.getLatestEmail();
+    expect(email).toBeDefined();
+    if (!email) return;
+
+    const otp = "000000";
+
+    await formPage.submitOTP(otp);
+
+    await expect(formPage.errorMessage).toBeVisible();
+  });
+
+  test("Resend OTP code", async ({ page }) => {
+    const testInbox = createTestInbox();
+    const formPage = new FormPage(page);
+
+    await formPage.load();
+
+    await formPage.selectAuthMethod("otp_via_email");
+    await formPage.enterEmail(testInbox.email);
+    await formPage.submitInitialForm();
+
+    const email = await testInbox.getLatestEmail();
+    expect(email).toBeDefined();
+    if (!email) return;
+
+    await formPage.promptCTA(); // resend OTP
+    await expect(formPage.emailIcon).toBeVisible();
```

Comment:
Similar question here, should we await for the second email and make sure a new code was delivered and you could log in? This was broken at least once so IMO it would make sense to do it.